tools/packaging/common/istio-iptables.sh

```diff
@@ -410,7 +410,7 @@ fi
 if [ -z "${DISABLE_REDIRECTION_ON_LOCAL_LOOPBACK-}" ]; then
   # Redirect app calls back to itself via Envoy when using the service VIP or endpoint
   # address, e.g. appN => Envoy (client) => Envoy (server) => appN.
-  iptables -t nat -A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -j ISTIO_REDIRECT
+  iptables -t nat -A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -j ISTIO_IN_REDIRECT
```

Comment:
@rshriram This line is the core change in this PR :)
This rule is supposed to  the case that client envoy send to server envoy. 
istio should capture this packet by redirect the packet to inbound listener(ISTIO_IN_REDIRECT stands for istio_*inbound*_redirect. 
If the inbound and outbound listener were not split, or inbound listener and outbound listener are equivalent(in/out listener are almost the same before introducing filterchains in virtual in/out bound listeners #15713), this bug is not hurting.

Reply:
Got it.